client/src/App.js

```diff
@@ -1,116 +1,185 @@
+import { useState, useEffect } from 'react';
+import { Switch, Route } from 'react-router-dom';
 import Home from './Pages/Home';
-import Watchlist from './Pages/Watchlist';
-import Friends from './Pages/Friends';
 import Search from './Pages/Search';
+import axios from './services/axios';
+import Friends from './Pages/Friends';
+import Ratings from './Pages/Ratings';
+import Profile from './Pages/Profile';
+import Watchlist from './Pages/Watchlist';
+import Favorites from './Pages/Favorites';
+import Sidebar from './components/Sidebar';
+import requests from './services/requests';
 import Banner from '../src/components/Banner';
-import Navigation from '../src/components/Navigation';
-import Store from './Store';
 import Searchbar from './components/Searchbar';
+import FriendsCards from './Pages/FriendsCards';
+import ScrollToTop from './services/ScrollToTop';
+import Navigation from '../src/components/Navigation';
+import { useLocalStorage } from '../src/hooks/useLocalStorage';
 
 function App() {
-  const [watchlist, setWatchList] = useLocalStorage('Watchlist', []);
-  const [search, setSearch] = useState([]);
   const [query, setQuery] = useState('');
+  const [open, setOpen] = useState(false);
+  const [search, setSearch] = useState([]);
+  const [show, handleShow] = useState(false);
   const [isLoading, setIsLoading] = useState(true);
-  const baseUrl = 'https://api.themoviedb.org/3/';
-  const APIKEY = process.env.REACT_APP_APIKEY;
+  const [watchlist, setWatchList] = useLocalStorage('Watchlist', []);
+  const [favorites, setFavorites] = useLocalStorage('Favorites', []);
+
+  let fetchUrl;
+  query === ''
+    ? (fetchUrl = requests.fetchTrending)
+    : (fetchUrl = `${requests.fetchSearch}${query}&page=1&include_adult=false`);
 
   useEffect(() => {
-    const fetchSearch = async () => {
+    async function fetchSearch() {
       setIsLoading(true);
-      const result = await axios(
-        `${baseUrl}search/movie?api_key=${APIKEY}&language=en-US&query=${query}&page=1&include_adult=false`
-      );
-      setSearch(result.data.results);
+      const request = await axios.get(fetchUrl);
+      setSearch(request.data.results);
       setIsLoading(false);
-    };
-    query !== '' ? fetchSearch() : setSearch([]);
-  }, [query, APIKEY]);
+      return request;
+    }
+
+    fetchSearch();
+  }, [query, fetchUrl]);
 
   const isOnWatchList = (movieToAdd) =>
     watchlist.some((movie) => movie.id === movieToAdd.id);
 
+  const isFavorite = (movieToAdd) =>
+    favorites.some((movie) => movie.id === movieToAdd.id);
+
   const addToWatchList = (movieToAdd) => {
     !isOnWatchList(movieToAdd)
       ? setWatchList([...watchlist, movieToAdd])
       : setWatchList(watchlist.filter((movie) => movie.id !== movieToAdd.id));
   };
 
+  const addToFavorites = (movieToAdd) => {
+    !isFavorite(movieToAdd)
+      ? setFavorites([...favorites, movieToAdd])
+      : setFavorites(favorites.filter((movie) => movie.id !== movieToAdd.id));
+  };
+
```

Comment:
You could maybe group together functions related to watchlist and to favorites so that it is easier to find the related function :) 
```suggestion
  const isOnWatchList = (movieToAdd) =>
    watchlist.some((movie) => movie.id === movieToAdd.id);

  const addToWatchList = (movieToAdd) => {
    !isOnWatchList(movieToAdd)
      ? setWatchList([...watchlist, movieToAdd])
      : setWatchList(watchlist.filter((movie) => movie.id !== movieToAdd.id));
  };

  const isFavorite = (movieToAdd) =>
    favorites.some((movie) => movie.id === movieToAdd.id);
    
  const addToFavorites = (movieToAdd) => {
    !isFavorite(movieToAdd)
      ? setFavorites([...favorites, movieToAdd])
      : setFavorites(favorites.filter((movie) => movie.id !== movieToAdd.id));
  };

```